src/lib/components/SveltyPicker.svelte

```diff
@@ -348,17 +348,29 @@
     const innerDate = innerDates[0] || now;
     onDate(new CustomEvent(resolvedMode, {
       detail: {
+        dateIndex: 0,
         value: new Date(
           now.getFullYear(),
           now.getMonth(),
           now.getDate(),
-          innerDate.getHours(),
-          innerDate.getMinutes(),
+        	isRange ? 0 : innerDate.getHours(),
```

Comment:
in range mode we want the start of the day, and in non range mode, i did not want to interfere with the current behavior

Reply:
I agree